### DISCUSSION TITLE
虚拟人导致页面中某个View变黑

### DISCUSSION THREAD

**Discussion by iamlionel** [Tencent/libpag#2038] [General]
![WechatIMG5](https://github.com/Tencent/libpag/assets/11271012/1c170fdd-cf19-455f-8b23-07aa45544ed6)

这是布局：
`<?xml version="1.0" encoding="utf-8"?>
<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:clipChildren="false"
    android:clipToPadding="false">

    <TextView
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        android:layout_width="0.1dp"
        android:layout_height="0.1dp" />
    
    <ImageView
        android:id="@+id/iv_icon"
        android:layout_width="@dimen/fhd_412"
        android:layout_height="@dimen/fhd_83"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        tools:src="@drawable/icon_recommend" />

    <TextView
        android:id="@+id/tv_title"
        android:layout_width="@dimen/fhd_313"
        android:layout_height="wrap_content"
        android:layout_marginStart="@dimen/fhd_71"
        android:layout_marginBottom="@dimen/fhd_10"
        android:maxLength="5"
        android:maxLines="1"
        android:textSize="@dimen/fhd_61"
        android:textStyle="bold|italic"
        android:typeface="monospace"
        app:layout_constraintBottom_toBottomOf="@id/iv_icon"
        app:layout_constraintStart_toStartOf="@id/iv_icon"
        app:layout_constraintTop_toTopOf="@id/iv_icon"
        tools:text="大家都在看" />

    <LinearLayout
        android:layout_width="@dimen/fhd_280"
        android:layout_height="@dimen/fhd_48"
        android:layout_marginEnd="@dimen/fhd_20"
        android:layout_marginBottom="@dimen/fhd_30"
        android:background="@drawable/bg_text_hint"
        android:gravity="center_vertical"
        android:orientation="horizontal"
        android:paddingStart="@dimen/fhd_15"
        android:paddingEnd="@dimen/fhd_11"
        android:paddingBottom="@dimen/fhd_2"
        app:layout_constraintBottom_toTopOf="@id/rv_list"
        app:layout_constraintEnd_toEndOf="parent">

        <TextView
            android:id="@+id/tv_back"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:textColor="@color/white"
            android:textSize="@dimen/fhd_28"
            tools:text="按返回键退出" />

        <View
            android:layout_width="@dimen/fhd_2"
            android:layout_height="@dimen/fhd_30"
            android:layout_marginStart="@dimen/fhd_10"
            android:background="@drawable/twig" />

        <TextView
            android:id="@+id/tv_hide"
            android:layout_width="0dp"
            android:layout_height="wrap_content"
            android:layout_weight="1"
            android:gravity="end"
            android:textColor="@color/white"
            android:textSize="@dimen/fhd_28"
            app:layout_constraintBottom_toTopOf="@id/rv_list"
            tools:text="60秒" />

    </LinearLayout>

    <androidx.recyclerview.widget.RecyclerView
        android:id="@+id/rv_list"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_marginTop="@dimen/fhd_30"
        android:clipChildren="false"
        android:clipToPadding="false"
        android:orientation="horizontal"
        android:paddingStart="@dimen/fhd_20"
        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
        app:layout_constraintTop_toBottomOf="@id/iv_icon"
        tools:itemCount="5"
        tools:listitem="@layout/item_recommend" />

    <TextView
        android:id="@+id/tv_prompt"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginStart="@dimen/fhd_20"
        android:layout_marginTop="@dimen/fhd_38"
        android:layout_marginBottom="@dimen/fhd_35"
        android:maxLength="22"
        android:maxLines="1"
        android:textColor="@color/white"
        android:textSize="@dimen/fhd_34"
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toBottomOf="@id/rv_list"
        tools:text="按住【语音键】试试说：“我想看刘德华的电影”"
        tools:textColor="@color/black" />
</androidx.constraintlayout.widget.ConstraintLayout>`

在这个布局中的第一个子View偶现变黑的现象，如上图所示。初步分析是硬件加速导致的。目前在布局中加了一个0.1dp的TextView来屏蔽。麻烦看一下。

**Comment by kevingpqi123** [comment]
建议描述清楚问题：虚拟人与 PAGView、普通View之前是啥关系，如何作用和影响的, 这里通篇没有提到 PAG 的使用

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
